flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/api/bridge/java/internal/StreamTableEnvironmentImpl.java

```diff
@@ -115,8 +115,10 @@ public static StreamTableEnvironment create(
                 new ResourceManager(settings.getConfiguration(), userClassLoader);
         final ModuleManager moduleManager = new ModuleManager();
         final CatalogStore catalogStore =
-                TableFactoryUtil.findAndCreateCatalogStore(
-                        settings.getConfiguration(), userClassLoader);
+                settings.getCatalogStore() != null
```

Comment:
In this branch ,who will take care the lifecycle of CatalogStoreFactory, i.e open/close connection resources?